File: event-master/app/src/main/java/com/vinodpatildev/eventmaster/data/model/Notification.kt
package com.vinodpatildev.eventmaster.data.model

import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName


data class Notification(
    @SerializedName("db_version")
    val __v: Int,
    @PrimaryKey
    @SerializedName("_id")
    val _id: String,
    @SerializedName("adminId")
    val adminId: String,
    @SerializedName("body")
    val body: String,
    @SerializedName("channel")
    val channel: String,
    @SerializedName("createdAt")
    val createdAt: String,
    @SerializedName("eventId")
    val eventId: String,
    @SerializedName("priority")
    val priority: String,
    @SerializedName("timestamp")
    val timestamp: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("type")
    val type: String,
    @SerializedName("updatedAt")
    val updatedAt: String
)

File: backend/src/models/notification.ts
import { InferSchemaType, model, Schema } from "mongoose";

const notificationSchema = new Schema({
    adminId: { type: Schema.Types.ObjectId, required: true },
    title: { type: String, required: true },
    body: { type: String },
    type: {type:String},
    eventId: { type: String },
    timestamp: { type: Date, default: Date.now, required: true },
    priority: { type: String, enum: ["high", "normal", "low"] },
    channel: { type: String },
},{
    timestamps : true
});

type Notification = InferSchemaType<typeof notificationSchema>;

export default  model<Notification>('Notification', notificationSchema);
